pkg/controller/master/virtualmachine/vmi_network_controller.go

```diff
@@ -62,9 +70,62 @@ func (h *VMNetworkController) updateVMDefaultNetworkMacAddress(vmi *kubevirtv1.V
 		}
 	}
 
+	if err := h.regenerateControllerRevision(vmi, vm); err != nil {
```

Comment:
this issue only happens when we patch back the mac address since the vmi does not need to be regenerated.

For any other updates for which VMI needs to be recreated the revision ends up being in sync